tools/pd-ctl/pdctl/command/keyspace_command.go

```diff
@@ -15,33 +15,278 @@
 	if err != nil {
-		cmd.Printf("Failed to get the keyspace information: %s\n", err)
+		cmd.PrintErrln("Failed to get the keyspace information: ", err)
+		return
+	}
+	cmd.Println(resp)
+}
+
+func newCreateKeyspaceCommand() *cobra.Command {
+	r := &cobra.Command{
+		Use:   "create <keyspace-name> [flags]",
+		Short: "create a keyspace",
+		Run:   createKeyspaceCommandFunc,
+	}
+	r.Flags().StringSlice(nmConfig, nil, "keyspace configs for the new keyspace\n"+
+		"specify as comma separated key value pairs, e.g. --config k1=v1,k2=v2")
+	return r
+}
+
+func createKeyspaceCommandFunc(cmd *cobra.Command, args []string) {
+	if len(args) != 1 {
+		cmd.Usage()
+		return
+	}
+
+	configPairs, err := cmd.Flags().GetStringSlice(nmConfig)
+	if err != nil {
+		cmd.PrintErrln("Failed to parse flag: ", err)
+		return
+	}
+	config := map[string]string{}
+	for _, flag := range configPairs {
+		kvs := strings.Split(flag, ",")
+		for _, kv := range kvs {
+			pair := strings.Split(kv, "=")
+			if len(pair) != 2 {
+				cmd.PrintErrf("Failed to create keyspace: invalid kv pair %s\n", kv)
+				return
+			}
+			if _, exist := config[pair[0]]; exist {
+				cmd.PrintErrf("Failed to create keyspace: key %s is specified multiple times\n", pair[0])
+				return
+			}
+			config[pair[0]] = pair[1]
+		}
+	}
+	params := handlers.CreateKeyspaceParams{
+		Name:   args[0],
+		Config: config,
+	}
+	body, err := json.Marshal(params)
+	if err != nil {
+		cmd.PrintErrln("Failed to encode the request body: ", err)
+		return
+	}
+	resp, err := doRequest(cmd, keyspacePrefix, http.MethodPost, http.Header{}, WithBody(bytes.NewBuffer(body)))
+	if err != nil {
+		cmd.PrintErrln("Failed to create the keyspace: ", err)
+		return
+	}
+	cmd.Println(resp)
+}
+
+func newUpdateKeyspaceConfigCommand() *cobra.Command {
+	r := &cobra.Command{
+		Use:   "update-config <keyspace-name>",
+		Short: "update keyspace config",
+		Run:   updateKeyspaceConfigCommandFunc,
+	}
+	r.Flags().StringSlice(nmRemove, nil, "keys to remove from keyspace config\n"+
+		"specify as comma separated keys, e.g. --remove k1,k2")
+	r.Flags().StringSlice(nmUpdate, nil, "kv pairs to upsert into keyspace config\n"+
+		"specify as comma separated key value pairs, e.g. --update k1=v1,k2=v2")
+	return r
+}
+
+func updateKeyspaceConfigCommandFunc(cmd *cobra.Command, args []string) {
+	if len(args) != 1 {
+		cmd.Usage()
+		return
+	}
+	configPatch := map[string]*string{}
+	removeFlags, err := cmd.Flags().GetStringSlice(nmRemove)
+	if err != nil {
+		cmd.PrintErrln("Failed to parse flag: ", err)
+		return
+	}
+	for _, flag := range removeFlags {
+		keys := strings.Split(flag, ",")
+		for _, key := range keys {
+			if _, exist := configPatch[key]; exist {
+				cmd.PrintErrf("Failed to update keyspace config: key %s is specified multiple times\n", key)
+				return
+			}
+			configPatch[key] = nil
+		}
+	}
+	updateFlags, err := cmd.Flags().GetStringSlice(nmUpdate)
+	if err != nil {
+		cmd.Println("Failed to parse flag: ", err)
```

Comment:
I find in this PR, some use Println and the others use PrintErrln.

Reply:
Ok, that one is fixed.
The description of `PrintErrln` and `Println` state that one print to defined output and the other prints to defined err output, so I use `PrintErrln` whenever theres an error and `Println` for regular response.

But it seems that underlying implementations of `PrintErrln` and `Println` are the same, which is quite different from `Print` vs `PrintErr`, and they are indeed used quite interchangeably in other commands of pd-ctl.
https://github.com/spf13/cobra/blob/4dd4b25de38418174a6e859e8a32eaccca32dccc/command.go#L1369

Reply:
It's now fixed.

In general, I use `Println` for response, and `PrintErrln` for any error occurred.

I believe the version of cobra we used does not really make difference between PrintErrln and Println, despite the description of them, resulting in that other places of pd-ctl use them interchangeably (they have the same implementation anyway). 

[This is fixed in newer version of cobra](https://github.com/spf13/cobra/pull/894), and we probably need to do some cleanup when we update the reference.